extensions/wikia/UserProfilePageV3/UserIdentityBox.class.php

```diff
@@ -314,7 +317,10 @@ public function saveUserData($data) {
 						}
 					}
 
-					if ($option === 'gender') {
+					if ($option === 'hideEditsWikis') {
+						// hideEditsWikis is a preference, everything else is an attribute
+						$this->user->setGlobalPreference($option, $data->$option);
+					} elseif ($option === 'gender') {
 						$this->user->setGlobalAttribute(self::USER_PROPERTIES_PREFIX . $option, $data->$option);
```

Comment:
Not quite. It's a little confusing, and initially I actually updated line 216 to clear up the logic a bit, but undid it b/c I felt like I was changing too much :)

L216 is actually checking if the option is not "gender" or "birthday" and if that's true (eg the option is say, "location" or "occupation") then it gets it without the prefix. If it is gender or birthday, then it gets it with the prefix. I actually think it would be a lot clearer written like:

```php
if ( $option === 'gender' || $option === 'birthday' ) {
    $data[$key] = $this->user->getGlobalAttribute( self::USER_PROPERTIES_PREFIX . $key );
} else {
    $data[$key] = $this->user->getGlobalAttribute( $key );
}
```

I think I'll go ahead and update that.